### PR TITLE
feat: Forces http2 for bidirectional event streams

### PR DESCRIPTION
**Pull request by epau**
This PR is the corresponding aws-sdk-swift PR for https://github.com/awslabs/smithy-swift/pull/545 which forces http2 for bidirectional event streams. 

## Issue \#
https://github.com/awslabs/aws-sdk-swift/issues/870

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Tests/Core/AWSClientRuntimeTests/Sigv4/SigV4SigningTests.swift [hunk 1]
```diff
@@ -9,23 +9,13 @@ import AwsCommonRuntimeKit
 import ClientRuntime
 import XCTest
 
-@testable import AWSClientRuntime
+@testable @_spi(Internal) import AWSClientRuntime
```
Comment (jbelkins) [inline Tests/Core/AWSClientRuntimeTests/Sigv4/SigV4SigningTests.swift:12]: What SPI is this intended to access?  (I thought we had renamed the SPI that was previously named `Internal`)

File: Tests/Core/AWSClientRuntimeTests/EventStream/AWSMessageEncoderStreamTests.swift [hunk 2]
```diff
@@ -23,60 +24,71 @@ final class AWSMessageEncoderStreamTests: XCTestCase {
...
             .withCredentialsProvider(value: MyCustomCredentialsProvider(credentials: credentials))
             .build()
-
+        
         let messageSigner = AWSEventStream.AWSMessageSigner(encoder: messageEncoder) {
-            try await context.makeEventStreamSigningConfig()
+            return try await context.makeEventStreamSigningConfig()
         } requestSignature: {
             return context.getRequestSignature()
         }
```
Comment (jbelkins) [inline Tests/Core/AWSClientRuntimeTests/EventStream/AWSMessageEncoderStreamTests.swift:46]: lol I had not seen this "multiple trailing closures" syntax until now... TIL

File: IntegrationTests/Services/AWSTranscribeStreamingIntegrationTests/TranscribeStreamingTests.swift [hunk 3]
```diff
@@ -17,7 +17,7 @@ final class TranscribeStreamingTests: XCTestCase {
         let chunkSize = 4096
         let audioDataSize = audioData.count
 
-        let client = try TranscribeStreamingClient(region: "us-west-2")
+        let client = try await TranscribeStreamingClient(region: "us-west-2")
```
[the remark this answers is not in the document]
Comment (epau) [reply IntegrationTests/Services/AWSTranscribeStreamingIntegrationTests/TranscribeStreamingTests.swift:20]: yep! will update this when I update the base branch (870-event-streaming-impl), which is a bit out of date now.

File: Tests/Core/AWSClientRuntimeTests/Sigv4/SigV4SigningTests.swift (hunk 1 above)
Comment (epau) [reply Tests/Core/AWSClientRuntimeTests/Sigv4/SigV4SigningTests.swift:12]: On line 70, the file based config line `try! await credentialsProvider.configure(.init(fileBasedConfigurationStore: .init()))`

File: IntegrationTests/Services/AWSTranscribeStreamingIntegrationTests/TranscribeStreamingTests.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (jbelkins) [reply IntegrationTests/Services/AWSTranscribeStreamingIntegrationTests/TranscribeStreamingTests.swift:20]: Ah! I didn't see that this branch doesn't merge to `main`

**Pull request merged**